src/generic/datavgen.cpp

```diff
@@ -2084,11 +2084,61 @@ bool wxDataViewMainWindow::EnableDragSource( const wxDataFormat &format )
 
 bool wxDataViewMainWindow::EnableDropTarget( const wxDataFormat &format )
 {
-    m_dropFormat = format;
-    m_dropEnabled = format != wxDF_INVALID;
+    if (format == wxDF_INVALID)
+        return false;
+
+    wxDataObjectComposite *dataObject;
+    wxDataViewDropTarget *dropTarget = static_cast<wxDataViewDropTarget*>(GetDropTarget());
+    if (dropTarget)
+        dataObject = static_cast<wxDataObjectComposite*>(dropTarget->GetDataObject());
```

Comment:
```suggestion
    {
        dataObject = static_cast<wxDataObjectComposite*>(dropTarget->GetDataObject());
    }
```
Just for consistency with the `else` branch,